docs/my-website/docs/proxy/guardrails/acuvity_ai.md

```diff
@@ -0,0 +1,271 @@
+
+## Integration Stages
+
+1. **Pre LLM API Call**
+
+   - **Data Loss Prevention (DLP) and Exploit Prevention:** Analyze incoming data for sensitive or malicious content. Based on predefined policies, the system can redact sensitive information or reject the request before it reaches the LLM.
+
+2. **During LLM API Call**
+
+   - **Policy Enforcement:** Evaluate requests in real-time. If the content matches specific criteria, the system rejects the request. Note: Redaction is not feasible at this stage.
+
+3. **Post LLM API Call**
+
+   - **Data Loss Prevention (DLP) and Exploit Prevention:** Examine the LLM's output for sensitive or malicious content. Depending on the policies, the system can redact sensitive information or reject the response before delivery.
+
+By integrating Acuvity's security functions at these critical points, users can ensure end-to-end protection throughout the LLM API call lifecycle.
+
+
+## 3. Define Guardrails in Your LiteLLM `config.yaml`
+
+### **1. Pre-Call: Detect and Redact PII**
+
+Add the **PII Detection** guardrail to your **Pre LLM API Call** configuration.
+
+**Redaction vs Detection:**
+- **Redacted PII** → The sensitive data is masked before being sent to the LLM (e.g., replacing emails and SSNs with `XXXXXXXX`).
+- **Detected PII** → The system identifies sensitive data but does not modify it. Detection alone does not prevent the request from being processed.
+
+> **How to configure Redaction vs Detection in `config.yaml`:**
+> - Use `redact: true` under `matches` to **redact** specific PII types.
+> - If `redact` is omitted, the system **only detects** the PII without modifying the request.
+
+> **Example:**
+> ```yaml
+> guardrails:
+>   - name: pii_detector
+>     matches:
+>       email_address:
+>         redact: true  # Email addresses will be redacted
+>       ssn:
+>         redact: true  # SSNs will be redacted
+>       person:         # Names will only be detected, not redacted
+> ```
+
+> **Note:** Redacting PII allows the request to proceed with masked data, while detecting PII without redaction simply rejects the call.
+
+
+### **2. During-Call: Detect Prompt Injection**
+
+Enable **Prompt Injection Detection** for your **During LLM API Call** configuration.
+
+### **3. Post-Call: Monitor Responses for Security Issues**
+
+Configure **Post LLM API Call** guardrails to filter inappropriate or malicious responses.
+
+#### For more details on using Acuvity with guardrails, visit [Acuvity Documentation](https://docs.acuvity.ai).
+
+Update your LiteLLM `config.yaml` file to include Acuvity guardrails:
+
+### config.yaml
+```yaml
+model_list:
+  - model_name: gpt-4
+    litellm_params:
+      model: openai/gpt-4
+      api_key: os.environ/OPENAI_API_KEY
+
+guardrails:
+  - guardrail_name: "acuvity-pii-detector"
```

Comment:
maybe we should call this `acuvity-pii-redaction`? As this is not a preventative guardrail like that. (Same naming a bit more above).

Reply:
done.